src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Legacy/RoslynCSharpTokenizer.cs

```diff
@@ -134,6 +134,7 @@ protected override StateResult Dispatch()
                 return StarAfterRazorCommentBody();
             case RoslynCSharpTokenizerState.AtTokenAfterRazorCommentBody:
                 Debug.Assert(_currentCSharpTokenTriviaEnumerator is not null);
+                _isOnlyWhitespaceOnLine = false;
```

Comment:
The issue here is that we'd process through the `@* .... *@`, and we wouldn't have reset `_isOnlyWhitespaceOnLine` to false. So when we took a snapshot of the current state when getting the `<` token, we'd record that the current state was `true`. Then, when the C# parser decided that this was HTML, and put the token back, an assert would fail because, in fact, there was other things than whitespace on the current line.